CMakeLists.txt

```diff
@@ -71,9 +71,12 @@ add_library(zephyr_interface INTERFACE)
 # flags that come with zephyr_interface.
 zephyr_library_named(zephyr)
 
-zephyr_include_directories(
+zephyr_system_include_directories(
```

Comment:
As you correctly state in the commit message, this will make some compilers / platforms skip certain warnings.
But warnings may also provide useful information, so before including this change could you please raise an issue with the warnings you see ?

Maybe this can be solved differently.